### PR TITLE
HDDS-10774. Show deleted hsync keys in ListOpenFile CLI.

### PR DESCRIPTION
**Pull request by ashishkumar50**
## What changes were proposed in this pull request?

ListOpenFile currently displays all the keys present in openFile/Key table. In some cases when hsync keys are deleted but those keys may still present in openFileTable for some duration before openKeyCleanupService deletes them permanently. To show those deleted keys in this PR we are adding one extra column in CLI to display deleted information.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10774

## How was this patch tested?

Integration test

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneShellHA.java [hunk 1]
```diff
@@ -699,6 +702,92 @@ public void testAdminCmdListOpenFiles()
...
+      // Wait for DB flush
+      cluster.getOzoneManager().awaitDoubleBufferFlush();
+
+      // hsync last key
+      streams[numKeys - 1].hsync();
+      // Wait for flush
+      cluster.getOzoneManager().awaitDoubleBufferFlush();
+      final String[] args = new String[] {"om", "lof",
+          "--service-id", omServiceId,
+          "-p", pathToBucket};
```
Comment (smengcl) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneShellHA.java:757]: Would you verify JSON output as well?

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/ListOpenFilesSubCommand.java
Comment (smengcl) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/ListOpenFilesSubCommand.java:1]: We might want to have an option to filter out the deleted keys. To make it easier, we can implement the filtering on the client side (the filtered list size maybe less than `--length` but it is fine).

**Review by smengcl — COMMENTED** [review]
Thanks @ashishkumar50 for the patch.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneShellHA.java (hunk 1 above)
Comment (ashishkumar50) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneShellHA.java]: Done

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/ListOpenFilesSubCommand.java
Comment (ashishkumar50) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/ListOpenFilesSubCommand.java:1]: Done

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/ListOpenFilesSubCommand.java [hunk 2]
```diff
@@ -66,6 +66,11 @@ public class ListOpenFilesSubCommand implements Callable<Void> {
       description = "Format output as JSON")
   private boolean json;
 
+  @CommandLine.Option(names = { "--hide-deleted" },
+      defaultValue = "false",
```
Comment (ChenSammi) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/ListOpenFilesSubCommand.java:70]: I would suggest we hide the deleted by default.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/ListOpenFilesSubCommand.java [hunk 3]
```diff
@@ -132,7 +140,7 @@ private void printOpenKeysList(ListOpenFilesResult res) {
     if (startItem != null && !startItem.isEmpty()) {
       msg += "\nafter continuation token:\n  " + startItem;
     }
-    msg += "\n\nClient ID\t\tCreation time\tHsync'ed\tOpen File Path";
+    msg += "\n\nClient ID\t\tCreation time\tHsync'ed\tDeleted\t\tOpen File Path";
```
Comment (ChenSammi) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/ListOpenFilesSubCommand.java:143]: If deleted is hide, we shall remove this Deleted column too.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/ListOpenFilesSubCommand.java (hunk 2 above)
Comment (ChenSammi) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/ListOpenFilesSubCommand.java:70]: Maybe we can call this --show-deleted.

Comment (ashishkumar50) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/ListOpenFilesSubCommand.java:70]: Done

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/ListOpenFilesSubCommand.java (hunk 3 above)
Comment (ashishkumar50) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/ListOpenFilesSubCommand.java]: Done

**Review by ChenSammi — APPROVED** [review]
Thanks @ashishkumar50 . The last patch LGTM.

**Pull request merged**